test/test_torch.py

```diff
@@ -5190,6 +5190,27 @@ def test_dot_data_use(self):
             model.weight.data = weight
             out = model(input)
 
+    def test_cpp_stacktraces(self):
+        python_error_msg = "Trying to create tensor with negative dimension"
+        cpp_stack_msg = "Exception raised from check_size_nonnegative at"
+
+        def check_error(has_cpp_stack_trace):
+            raised = False
+            try:
+                torch.rand(-1)
+            except Exception as e:
+                raised = True
+                err_msg = str(e)
+                self.assertTrue(python_error_msg in err_msg)
+                self.assertTrue((cpp_stack_msg in err_msg) == has_cpp_stack_trace)
+
+        orig = torch.utils.debug.is_cpp_stacktraces_enabled()
+        torch.utils.debug.set_cpp_stacktraces_enabled(False)
+        check_error(False)
+        torch.utils.debug.set_cpp_stacktraces_enabled(True)
+        check_error(True)
+        torch.utils.debug.set_cpp_stacktraces_enabled(orig)
```

Comment:
If this code raises an exception (because the assert fails) you will leave the global variable here toggled to the test state (failing to reset it). Use a finally block here.